### PR TITLE
feat: lemmas about Rat.instIntCast

### PR DESCRIPTION
**Pull request by semorrison**
I need these lemmas without Mathlib.

### CODE REVIEW

File: Std/Data/Int/Lemmas.lean [hunk 1]
```diff
@@ -1415,3 +1415,20 @@ theorem mem_toNat' : ∀ (a : Int) (n : Nat), toNat' a = some n ↔ a = n
...
+
+@[simp] theorem natCast_add (a b : Nat) : ((a + b : Nat) : Int) = (a : Int) + (b : Int) := by
+  -- Note this only works because of local simp attributes in this file,
+  -- so it still makes sense to tag the lemmas with `@[simp]`.
+  simp
+
+@[simp] theorem natCast_mul (a b : Nat) : ((a * b : Nat) : Int) = (a : Int) * (b : Int) := by
+  simp
+
+end Int
```
Comment (digama0) [inline Std/Data/Int/Lemmas.lean:1434]:
```suggestion
```

File: Std/Data/Rat/Lemmas.lean [hunk 2]
```diff
@@ -317,3 +318,32 @@ theorem ofScientific_false_def : Rat.ofScientific m false e = (m * 10 ^ e : Nat)
...
+@[simp] theorem intCast_sub (a b : Int) : ((a - b : Int) : Rat) = (a : Rat) - (b : Rat) := by
+  rw [sub_def]
+  ext <;> simp [normalize_eq]
+
+@[simp] theorem intCast_mul (a b : Int) : ((a * b : Int) : Rat) = (a : Rat) * (b : Rat) := by
+  rw [mul_def]
+  ext <;> simp [normalize_eq]
+
+example (a b : Nat) : ((a + b : Nat) : Rat) = (a : Rat) + (b : Rat) := by simp
+example (a b : Nat) : ((a * b : Nat) : Rat) = (a : Rat) * (b : Rat) := by simp
```
Comment (digama0) [inline Std/Data/Rat/Lemmas.lean:349]:
```suggestion
```

File: Std/Data/Int/Lemmas.lean [hunk 3]
```diff
@@ -1415,3 +1415,18 @@ theorem mem_toNat' : ∀ (a : Int) (n : Nat), toNat' a = some n ↔ a = n
 @[simp] theorem toNat_neg_nat : ∀ n : Nat, (-(n : Int)).toNat = 0
   | 0 => rfl
   | _+1 => rfl
+
+-- The following lemmas are later subsumed by e.g. `Nat.cast_add` and `Nat.cast_mul` in Mathlib
+-- but it is convenient to have these earlier, for users who only need `Nat` and `Int`.
```
Comment (eric-wieser) [inline Std/Data/Int/Lemmas.lean:1420]: This is probably useful to show in the docs?
```suggestion
/-!
The following lemmas are later subsumed by e.g. `Nat.cast_add` and `Nat.cast_mul` in Mathlib
but it is convenient to have these earlier, for users who only need `Nat` and `Int`.
-/
```

Comment (eric-wieser) [reply Std/Data/Int/Lemmas.lean:1420]: (same below)

File: Std/Data/Rat/Lemmas.lean [hunk 4]
```diff
@@ -317,3 +318,29 @@ theorem ofScientific_false_def : Rat.ofScientific m false e = (m * 10 ^ e : Nat)
 theorem ofScientific_def : Rat.ofScientific m s e =
     if s then mkRat m (10 ^ e) else (m * 10 ^ e : Nat) := by
   cases s; exact ofScientific_false_def; exact ofScientific_true_def
+
+@[simp] theorem intCast_den (a : Int) : (a : Rat).den = 1 := rfl
+
+@[simp] theorem intCast_num (a : Int) : (a : Rat).num = a := rfl
+
+-- The following lemmas are later subsumed by e.g. `Int.cast_add` and `Int.cast_mul` in Mathlib
+-- but it is convenient to have these earlier, for users who only need `Int` and `Rat`.
```
Comment (semorrison) [inline Std/Data/Rat/Lemmas.lean:327]:
```suggestion
/-!
The following lemmas are later subsumed by e.g. `Int.cast_add` and `Int.cast_mul` in Mathlib
but it is convenient to have these earlier, for users who only need `Int` and `Rat`.
-/
```

**Pull request merged**